Libplanet/Net/Swarm.cs

```diff
@@ -145,6 +145,13 @@ public partial class Swarm : ICollection<Peer>, IDisposable
             }
         }
 
+        /// <summary>
+        /// The <see cref="EventHandler" /> called when the different version of
+        /// <see cref="Peer" /> is discovered.
+        /// </summary>
+        public event EventHandler<DifferentProtocolVersionEventArgs>
+            DifferentVersionPeerDiscovered;
```

Comment:
It would be better if the `Swam` constructor can take this as an optional parameter.

Reply:
@kijun @longfin Which word is better, discovered vs. encountered?  Or any other word?

Reply:
Hmm maybe this is not an option, but ProtocolVersionConflict sounds more concise... thoughts?

Reply:
Even if a peer have a different protocol we still treat them as compatible, right?  It's one of options game programmers have.  So naming it “conflict” feels like oversimplified to me.

Reply:
Ah, I wasn't aware of this. I'd say then `Encountered` is slightly better since we're not actively trying to find peers with different protocol version.

Reply:
Ok, So I'll change `Discovered` to `Encountered`.